stubs/CoreGenericFunctions.phpstub

```diff
@@ -1086,6 +1086,31 @@ function chunk_split(string $str, int $chunklen = 76, string $end= ''): string
 {
 }
 
+/**
+ * @template TInput as \Countable|\SimpleXMLElement|\ResourceBundle|array
+ *
+ * @param TInput $value
+ * @param 0|1 $mode
+ *
+ * @return (
+ *          TInput is array<empty, empty>
+ *          ? 0
+ *          : (TInput is array{0: mixed, 1: mixed}
+ *              ? 2
+ *              : (TInput is non-empty-array
+ *                  ? positive-int
+ *                  : (TInput is array
+ *                      ? positive-int|0
+ *                      : int
+ *                  )
+ *              )
+ *          )
+ * )
+ */
+function count(\Countable|\SimpleXMLElement|\ResourceBundle|array $value, int $mode = 0): int
```

Comment:
Tempted to mark this `@psalm-pure`, but `@psalm-pure` only applies to `array` as input :-\

Reply:
Need feedback based on findings at https://github.com/vimeo/psalm/pull/4999#issuecomment-759444684